Source/LinqToDB.EntityFrameworkCore/LinqToDBForEFTools.cs

```diff
@@ -558,5 +567,71 @@ public static bool EnableChangeTracker
 			set => Implementation.EnableChangeTracker = value;
 		}
 
+		private static void AddInterceptorsToDataContext(DbContext efContext, IDataContext dc)
+		{
+			var contextOptions = ((IInfrastructure<IServiceProvider>)efContext.Database)?
+				.Instance?.GetService(typeof(IDbContextOptions)) as IDbContextOptions;
+			
+			AddInterceptorsToDataContext(contextOptions, dc);
+		}
+
+		private static void AddInterceptorsToDataContext(IDbContextOptions? contextOptions,
+			IDataContext dc)
+		{
+			var registeredInterceptors = contextOptions?.GetLinq2DbInterceptors();
+
+			if (registeredInterceptors?.Any() == true
+				&& dc != null )
+			{
+				var commandInterceptable = dc as IInterceptable<ICommandInterceptor>;
+				var connectionInterceptable = dc as IInterceptable<IConnectionInterceptor>;
+				var entityServiceInterceptable = dc as IInterceptable<IEntityServiceInterceptor>;
+				var dataContextInterceptable = dc as IInterceptable<IDataContextInterceptor>;
+
+				foreach (var interceptor in registeredInterceptors)
+				{
+					string interceptorTypeFullName = interceptor.GetType().FullName;
+					var existingMainInterceptors = new List<IInterceptor>();

```

Comment:
Why this code exists, why not just add interceptors to the context?

Reply:
This is in a comment in code - basically to prevent duplicated interceptors in pipeline. If you add multiple time the same interceptor, then its logic will fire multiple times for no reason.

Reply:
In that case:
1. This is user's issue. We shouldn't encourage bad code. Even more this way we are hiding configuration bug in user's code. I will not appreciate such behavior from library I use.
2. This check is not correct as it basically doesn't allow user to add more than one interceptor with the same type. If interceptor contains configurable options, it will prevent user from adding multiple interceptors with different options.

Reply:
Ok, I removed that check then.